core/capabilities/ccip/ccipevm/helpers.go

```diff
@@ -43,6 +51,54 @@ func decodeExtraArgsV1V2(extraArgs []byte) (gasLimit *big.Int, err error) {
 	return ifaces[0].(*big.Int), nil
 }
 
+func DecodeDestExecDataToMap(DestExecData cciptypes.Bytes) (map[string]interface{}, error) {
+	destGasAmount, err := abiDecodeUint32(DestExecData)
+	if err != nil {
+		return nil, fmt.Errorf("decode dest gas amount: %w", err)
+	}
+
+	return map[string]interface{}{
+		evmDestExecDataKey: destGasAmount,
+	}, nil
+}
+
+func DecodeExtraArgsToMap(extraArgs []byte) (map[string]any, error) {
+	if len(extraArgs) < 4 {
+		return nil, fmt.Errorf("extra args too short: %d, should be at least 4 (i.e the extraArgs tag)", len(extraArgs))
+	}
+
+	var method string
+	var extraByteOffset int
+	switch string(extraArgs[:4]) {
+	case string(evmExtraArgsV1Tag):
+		// for EVMExtraArgs, the first four bytes is the method name
+		method = evmV1DecodeName
+		extraByteOffset = 4
+	case string(evmExtraArgsV2Tag):
+		method = evmV2DecodeName
+		extraByteOffset = 4
+	case string(svmExtraArgsV1Tag):
+		// for SVMExtraArgs there's the four bytes plus another 32 bytes padding for the dynamic array
+		// https://github.com/smartcontractkit/chainlink/blob/33c0bda696b0ed97f587a46eacd5c65bed9fb2c1/contracts/src/v0.8/ccip/libraries/Client.sol#L57
+		method = svmV1DecodeName
+		extraByteOffset = 36
```

Comment:
I still don't see why this is 36 instead of 4. It looks like it's encoded the same way as EVM args are encoded: https://github.com/smartcontractkit/chainlink/blob/33c0bda696b0ed97f587a46eacd5c65bed9fb2c1/contracts/src/v0.8/ccip/libraries/Client.sol#L74C35-L74C62

Reply:
https://chainlink-core.slack.com/archives/C0779R3AH8R/p1737837136262789?thread_ts=1737484810.494739&cid=C0779R3AH8R